### ISSUE TITLE
No-member error not raised when class is forced inside function

### ISSUE THREAD

**Issue by Fvegini** [PyCQA/pylint#7135]
### Bug description

    Class Dummyclass(object):

        __slots__ = ['myvariable']

        def __init__(self):
            self.myvariable = 1

    def main(other_dummy: Dummyclass):

        dummy = Dummyclass()
        dummy.doesnotexist = 4 #This raises an error assign-non-slot
        temp1 = dummy.doesnotexist  #This raises an error no-member
        other_dummy.doesnotexist = 4 #This is not raising error
        temp2 = other_dummy.doesnotexist #This is not raising error

    new_dummy = Dummyclass()
    main(new_dummy)

Pylint it not raising errors then the other_dummy variable is trying to access a no-member variable.
This may be related to the fact that the typing is just a hint, not a mandatory thing. But the code breaks on those lines and not a single warning is shown

### Command used

```shell
pylint linter_test.py
```

### Pylint output

```shell
linter_test.py:10:4: E0237: Assigning to attribute 'doesnotexist' not defined in class slots (assigning-non-slot)
linter_test.py:11:12: E1101: Instance of 'Dummyclass' has no 'doesnotexist' member (no-member)
linter_test.py:11:4: W0612: Unused variable 'temp1' (unused-variable)
linter_test.py:13:4: W0612: Unused variable 'temp2' (unused-variable)
```

### Expected behavior

Lines 12 and 13 should yell the same errors that lines 10 and 11 are yelling.
The code totally breaks in lines 12 and 13, so I believe the user should be yelled (or at least warned)

### Pylint version

```shell
pylint 2.14.4
astroid 2.11.6
Python 3.9.7 (default, Feb  9 2022, 13:01:52)
```

### OS / Environment

Ubuntu

**Comment by Fvegini** [comment]
I made a test with mypy and it signalizes the errors:

    mypy linter_test.py 
    linter_test.py:10: error: "Dummyclass" has no attribute "doesnotexist"
    linter_test.py:11: error: "Dummyclass" has no attribute "doesnotexist"
    linter_test.py:12: error: "Dummyclass" has no attribute "doesnotexist"
    linter_test.py:13: error: "Dummyclass" has no attribute "doesnotexist"

**Comment by Pierre-Sassoulas** [comment]
Right now pylint do not check type. Closing as duplicate of  https://github.com/PyCQA/pylint/issues/4813
